src/CSS.cpp

```diff
@@ -13,36 +13,33 @@ namespace CSS
     {
         sProvider = gtk_css_provider_new();
 
-        struct CSSDir
-        {
-            std::string xdgEnv;
-            std::string fallbackPath;
-            std::string relPath;
-        };
-        std::string home = getenv("HOME");
-        std::array<CSSDir, 4> locations = {
-            CSSDir{"XDG_CONFIG_HOME", home + "/.config",      "/gBar/style.css"}, // Local config
-            CSSDir{"XDG_DATA_HOME",   home + "/.local/share", "/gBar/style.css"}, // local user install
-            CSSDir{"",                "/usr/local/share",     "/gBar/style.css"}, // local all install
-            CSSDir{"",                "/usr/share",           "/gBar/style.css"}, // package manager all install
-        };
+        std::vector<std::string> locations;
+        const char * env;
+
+        env = std::getenv("XDG_CONFIG_HOME");
+        ASSERT(env, "XDG_CONFIG_HOME is not set!");
+        locations.push_back(env);
+        
+        env = std::getenv("XDG_DATA_DIRS");
+        ASSERT(env, "XDG_DATA_DIRS is not set!");
+        std::string xdgDataDirs(env);
```

Comment:
Creating a string just for the stringstream is unnecessary, since stringstream has it's own copy for the string and can accept char* as input (e.g. ```std::stringstream ss(env);```).